### PR TITLE
257 minimal demo data lbk and abcd

### LINKED ISSUE #257: feat: `loanbook_demo` drops columns that are not crucial to `r2dii.analysis` functions
Relates to feedback from a bank. 
Since our data preparation guidance is to prepare input data exactly as `loanbook_demo`, we should either: 
* Keep all columns that are currently in `loanbook_demo`, but indicate which are actually crucial to `target_market_share` and `target_sda`
OR
* Remove the non-crucial columns

[AB#10167](https://dev.azure.com/RMI-PACTA/d77fc8f8-abac-42ae-9e31-846f9901290d/_workitems/edit/10167)

### LINKED ISSUE #372: upkeep: in `abcd_demo`, rename `abcd_timestamp` to `ald_timestamp`
In #315 , the `abcd_demo` dataset had the `ald_timestamp` column renamed to `abcd_timestamp`. 

Unfortunately, the abcd data as Asset Impact distributes it still contains the name `ald_timestamp`. 

When pressed on this topic, Asset Impact was worried the breaking change of changing a column name might impact users existing workflows, and were hesitant to do so. 

We should align the column name with what is in the expected dataset. 

Alternatively, we could remove the column entirely (relates to #257 ), as I don't believe it is a critical column in any of the `pacta.*` or `r2dii.*` code:
https://github.com/search?q=org%3ARMI-PACTA%20ald_timestamp&type=code
https://github.com/search?q=org%3ARMI-PACTA+abcd_timestamp&type=code

FYI @jacobvjk

### PR DESCRIPTION
**Pull request by jacobvjk**
closes #257 
closes #372 

- `abcd_demo` drops columns `abcd_timestamp` (not needed for calculation) and `country_of_domicile` (not part of the P4B format ABCD)
- `loanbook_demo` drops columns `id_intermediate_parent_1`, `name_intermediate_parent_1`,, `sector_classification_input_type`, `fi_type`, `flag_project_finance_loan`, and `name_project`, none of which are needed for the calculation
- `data_dictionary` is adjusted accordingly and drops the mention of these variables
- snaps are updated

**NOTE:**
- we may want to keep the variables that were removed from the demo data sets in the `data_dictionary`. If so, test expectations need to be adjusted

The new demo data sets generate the following demo output
``` r
library(dplyr)
#> 
#> Attaching package: 'dplyr'
#> The following objects are masked from 'package:stats':
#> 
#>     filter, lag
#> The following objects are masked from 'package:base':
#> 
#>     intersect, setdiff, setequal, union
library(r2dii.data)
library(r2dii.match)
library(r2dii.analysis)
library(r2dii.plot)

names(loanbook_demo)
#>  [1] "id_loan"                               
#>  [2] "id_direct_loantaker"                   
#>  [3] "name_direct_loantaker"                 
#>  [4] "id_ultimate_parent"                    
#>  [5] "name_ultimate_parent"                  
#>  [6] "loan_size_outstanding"                 
#>  [7] "loan_size_outstanding_currency"        
#>  [8] "loan_size_credit_limit"                
#>  [9] "loan_size_credit_limit_currency"       
#> [10] "sector_classification_system"          
#> [11] "sector_classification_direct_loantaker"
#> [12] "lei_direct_loantaker"                  
#> [13] "isin_direct_loantaker"
# "lei_direct_loantaker" is optional, but may be used as a `join_id`
# "isin_direct_loantaker" is optional, but may be used as a `join_id`

names(abcd_demo)
#>  [1] "company_id"           "name_company"         "lei"                 
#>  [4] "sector"               "technology"           "production_unit"     
#>  [7] "year"                 "production"           "emission_factor"     
#> [10] "plant_location"       "is_ultimate_owner"    "emission_factor_unit"

matched <- loanbook_demo |> 
  match_name(abcd_demo) |> 
  prioritize()

matched |> 
  target_market_share(abcd_demo, scenario_demo_2020, region_isos_demo) |> 
  filter(sector == "power", region == "global", technology == "renewablescap") |> 
  prep_trajectory() |> 
  plot_trajectory()
```

``` r

matched |> 
  target_sda(
    abcd_demo, 
    co2_intensity_scenario_demo, 
    region_isos = region_isos_demo
  ) |> 
  filter(sector == "cement", region == "global") |> 
  prep_emission_intensity() |> 
  plot_emission_intensity()
#> Warning: Removing rows in abcd where `emission_factor` is NA
```

<sup>Created on 2024-04-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

### CODE REVIEW

**Comment by jdhoffa** [comment]
Just noting that:
`loanbook_demo` drops columns `id_intermediate_parent_1`, `name_intermediate_parent_1` MIGHT affect some downstream vignettes in `r2dii.match`... not certain atm

File: data-raw/loanbook_demo.R [hunk 1]
```diff
@@ -9,24 +9,18 @@ source(file.path("data-raw", "utils.R"))
 path <- file.path("data-raw", "loanbook_demo.csv")
 loanbook_demo <- readr::read_csv(
   path,
-  col_types = readr::cols(
+  col_types = readr::cols_only(
     id_loan = "c",
     id_direct_loantaker = "c",
     name_direct_loantaker = "c",
-    id_intermediate_parent_1 = "c",
-    name_intermediate_parent_1 = "c",
```
Comment (jdhoffa) [inline data-raw/loanbook_demo.R:17]: I would test this very carefully with `r2dii.match`. 
e.g. 
* install the dev version of `r2dii.data` 
* run all tests in `r2dii.match`, try to build/ run all vignettes and `.Rmd` files, README, checks and examples, etc.

I have in my head for some reason that, while not a "crucial" column, we have certain code that expects these columns...

Comment (jacobvjk) [reply data-raw/loanbook_demo.R:17]: So I tested the following:
- installed `r2dii.data` dev version locally
- ran all tests in `r2dii.match`
  - made adjustments to test-helper of expected variables and snaps as seen here: https://github.com/RMI-PACTA/r2dii.match/pull/473
  - tests now pass
- adjusted vignette to clarify that `intermediate_parent_*` is always optional and not part of the demo data
- locally built all vignettes using `devtools::build_rmd("vignettes/*.Rmd")` based on https://r-pkgs.org/vignettes.html#sec-vignettes-workflow-writing
  - all vignettes successfully built locally

`intermediate_parent_*` is indeed used in other tests, but it is explicitly added via `fake_lbk()`, which is not affected by the changes in the demo data

**Comment by jacobvjk** [comment]
to feel safe about this, I would appreciate you checking the r2dii.match vignettes as well @jdhoffa in case you find time for that

File: data-raw/loanbook_demo.R (hunk 1 above)
Comment (jdhoffa) [reply data-raw/loanbook_demo.R:17]: 🚀

**Comment by jdhoffa** [comment]
Testing now!

**Review by jdhoffa — APPROVED** [review]
Looks great! 
I saw no issues when testing against this PR: https://github.com/RMI-PACTA/r2dii.match/pull/473

**Pull request merged**